app/components/DataTable/components/DetailDialog/__tests__/DetailDialog.test.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import { screen, render, fireEvent } from '@testing-library/react';
+
+import DetailDialog from '..';
+
+const mockData = {
+  detectedIn: 'DNA',
+  gene: {
+    name: 'EGFR',
+    isImportant: true,
+    innerObject: {
+      innerInnerValue: 'double nested',
+    },
+  },
+};
+
+const mockArrayData = {
+  arrayVal: ['test1', 'test2'],
+};
+
+const mockFilteredData = {
+  ident: '640aff03-402bdd42-dffa-623b',
+  svg: '<svg><g>image here</g></svg>',
+  svgTitle: 'test SVG',
+  image: 'test image',
+};
+
+const mockColumnMapping = {
+  detectedIn: 'Detected In',
+};
+
+describe('DetailDialog', () => {
+  test('It matches the snapshot', () => {
+    const { asFragment } = render(
+      <DetailDialog
+        isOpen
```

Comment:
a lot of these props can probably be put into a constant and reused throughout so you'd have an easier time

also seem to be missing props `onClose`